Documentation/rbd-mirroring.md

```diff
@@ -0,0 +1,374 @@
+
+To achieve RBD Mirroring, `csi-omap-generator` and `volume-replication`
+ containers need to be deployed in the RBD provisioner pods.
+
+* **Omap Generator**: Omap generator is a sidecar container that when
+ deployed with the CSI provisioner pod, generates the internal CSI
+ omaps between the PV and the RBD image. This is required as static PVs are
+ transferred across peer clusters in the DR use case, and hence
+ is needed to preserve PVC to storage mappings.
+
+* **Volume Replication Operator**: Volume Replication Operator is a
+ kubernetes operator that provides common and reusable APIs for
+ storage disaster recovery.
+ It is based on [csi-addons/spec](https://github.com/csi-addons/spec)
+ specification and can be used by any storage provider.
+ For more details, refer to [volume replication operator](https://github.com/csi-addons/volume-replication-operator).
+
+Execute the following steps on each peer cluster to enable the
+ OMap generator and Volume Replication sidecars:
+
+* Edit the `rook-ceph-operator-config` configmap and add the
+ following configurations
+
+```bash
+kubectl edit cm rook-ceph-operator-config -n rook-ceph
+```
+
+Add the following configuration if not present:
+
+```yaml
+data:
+  CSI_ENABLE_OMAP_GENERATOR: "true"
+  CSI_ENABLE_VOLUME_REPLICATION: "true"
+```
+
+* After updating the configmap with those settings, two new sidecars
+ should now start automatically in the CSI provisioner pod.
+* Repeat the steps on the peer cluster.
+
+## Volume Replication Custom Resources
+
+Volume Replication Operator follows controller pattern and provides extended
+APIs for storage disaster recovery. The extended APIs are provided via Custom
+Resource Definition (CRD). It provides support for two custom resources:
+
+* **VolumeReplicationClass**: *VolumeReplicationClass* is a cluster scoped
+resource that contains driver related configuration parameters. It holds
+the storage admin information required for the volume replication operator.
+
+* **VolumeReplication**: *VolumeReplication* is a namespaced resource that contains references to storage object to be replicated and VolumeReplicationClass
+corresponding to the driver providing replication.
+
+>:bulb: For more information, please refer to the
+> [volume-replication-operator](https://github.com/csi-addons/volume-replication-operator).
+
+## Enable mirroring on a PVC
+
+Below guide assumes that we have a PVC (rbd-pvc) in BOUND state; created using
+ *StorageClass* with `Retain` reclaimPolicy.
+
+```bash
+kubectl get pvc --context=cluster-1
+```
+
+>
+> ```bash
+> NAME      STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+> rbd-pvc   Bound    pvc-65dc0aac-5e15-4474-90f4-7a3532c621ec   1Gi        RWO            csi-rbd-sc   44s
+> ```
+
+### Create a Volume Replication Class CR
+
+In this case, we create a Volume Replication Class on cluster-1 ()
+
+```yaml
+cat <<EOF | kubectl --context=cluster-1 apply -f -
```

Comment:
Is this yaml in the examples folder? I don't see it. The preferred pattern for docs is to have this available as a yaml in the examples folder, then have them create the yaml from the examples. You can also show the yaml in the doc, but no need for actually generating the file since it's already in the examples folder.
```suggestion
```